api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java

```diff
@@ -191,6 +197,175 @@ public CohortDefinition getPatientsInArtWhoAbandoned() {
+
+    String query =
+        "SELECT most_recent.patient_id "
+            + "FROM   (SELECT p.patient_id, "
+            + "               o.value_datetime, "
+            + "               e.location_id, "
+            + "               e.encounter_id "
+            + "        FROM   patient p "
+            + "               inner join encounter e "
+            + "                       ON e.patient_id = p.patient_id "
+            + "               inner join obs o "
+            + "                       ON o.encounter_id = e.encounter_id "
+            + "        WHERE  p.voided = 0 "
+            + "               AND e.voided = 0 "
+            + "               AND o.voided = 0 "
+            + "               AND e.encounter_type = %s "
+            + "               AND o.value_datetime IS NOT NULL "
+            + "               AND o.concept_id = %s "
+            + "        UNION "
+            + "        SELECT p.patient_id, "
+            + "               o.value_datetime, "
+            + "               e.location_id, "
+            + "               e.encounter_id "
+            + "        FROM   patient p "
+            + "               inner join encounter e "
+            + "                       ON e.patient_id = p.patient_id "
+            + "               inner join obs o "
+            + "                       ON o.encounter_id = e.encounter_id "
+            + "        WHERE  p.voided = 0 "
+            + "               AND e.voided = 0 "
+            + "               AND o.voided = 0 "
+            + "               AND e.encounter_type IN ( %s, %s ) "
+            + "               AND o.concept_id = %s "
+            + "        UNION "
+            + "        SELECT p.patient_id, "
+            + "               o.value_datetime, "
+            + "               e.location_id, "
+            + "               e.encounter_id "
+            + "        FROM   patient p "
+            + "               inner join encounter e "
+            + "                       ON e.patient_id = p.patient_id "
+            + "               inner join obs o "
+            + "                       ON o.encounter_id = e.encounter_id "
+            + "        WHERE  p.voided = 0 "
+            + "               AND e.voided = 0 "
+            + "               AND o.voided = 0 "
+            + "               AND o.concept_id = %s "
+            + "               AND o.value_datetime IS NOT NULL "
+            + "               AND e.encounter_type = %s) most_recent "
+            + "WHERE  most_recent.value_datetime = (SELECT Max(value_datetime) "
+            + "                                     FROM   obs "
+            + "                                     WHERE  encounter_id = "
+            + "                                            most_recent.encounter_id) "
+            + "       AND most_recent.value_datetime < Date_add(:onOrBefore, interval 30 day) "
```

Comment:
@ynurmahomed so the specs says that the encounter date should be 30 days before the EndDate

"And add 30 days and this date should be < endDate"

But here we are adding 30days to the enddate - is this correct?

Reply:
This was supposed to be re-used from TX_CURR I forgot to remove it. Removed it now.

Reply:
Hmm ok but it is in the specs - did Pinki say we should just ignore?